### PR TITLE
BGM ,Server list and autoupdate updates

### PR DESCRIPTION
**Pull request by sleeser**
Menu song change from song.ogg to song.mp3 and was moved to the sounds folder.
Added disadvantage and advantage bgms (L0/L1 >= 2)
Multiple bgms are now supported for duels, the game will pick a random one from the bgm folder at the start of the duel.
An option to randomize the bgms in a duel so when one is finished a new one will be picked or looping the current playing bgm.
Server list is no longer hard coded, a new file name "servers.conf" was added to the conifg folder and will be downloaded when its missing. (the update client function was updated so it doenst delete files anymore)
The auto updater now generates the SHA-1 of existing local files in the config and cdb folders and compare that to what is in the update git and then updates the files with different SHA hashes.
(No longer generates "SHAs.txt" file and the updates folder is no longer needed)
Fixed bugs related to the game creating empty conf files and trying to read them afterwards resulting in infinite loops.
Fixed a bug where no deck is selected when starting a duel resulting in immediate disconnections.

### CODE REVIEW

File: Assets/SibylSystem/Program.cs [hunk 1]
```diff
@@ -1,4 +1,1105 @@
...
+//        setting.saveWhenQuit();
+//    }
+
+//    #endregion
+
+//    public static void gugugu()
+//    {
+//        PrintToChat(InterString.Get("非常抱歉，因为技术原因，此功能暂时无法使用。请关注官方网站获取更多消息。"));
+//    }
+//}
```
Comment (szefo09) [inline Assets/SibylSystem/Program.cs]: Why is all of this commented?

File: Assets/SibylSystem/Program.cs [hunk 2]
```diff
@@ -429,6 +1531,86 @@ private void UpdateClient()
...
+                        string sha = GetHashString(GetHash(bytes)).ToLower(); ;
+                        if (sha != file.sha)
+                        {
+                            toDownload.Add(file);
+                        }
+                    }
+                }
+                //foreach (string f in local)
+                //{
+                //    if (apiFromGit.FirstOrDefault(x => x.name == f) == null || f != apiFromGit.FirstOrDefault(x => x.name == f).name)
```
Comment (szefo09) [inline Assets/SibylSystem/Program.cs]: Why do you no longer want to remove unnecessary files?

File: Assets/SibylSystem/Program.cs (hunk 1 above)
Comment (sleeser) [reply Assets/SibylSystem/Program.cs]: Leftovers i forgot to delete

File: Assets/SibylSystem/Program.cs (hunk 2 above)
Comment (sleeser) [reply Assets/SibylSystem/Program.cs]: Because it deletes config.conf because it isnt on the udpate git

**Pull request closed without merging**